packages/typespec-azure-resource-manager/lib/models.tsp

```diff
@@ -15,18 +15,20 @@ namespace Azure.ResourceManager;
  * @template KeyName Override default key name of the resource.
  * @template SegmentName Override default segment name of the resource.
  * @template NamePattern The RegEx pattern of the name. Default is `^[a-zA-Z0-9-]{3,24}$`.
+ * @template NameType The type of the name property. Default type is string. However you can pass an union with string values.
  */
 model ResourceNameParameter<
   Resource extends Foundations.Resource,
   KeyName extends valueof string = "",
   SegmentName extends valueof string = "",
-  NamePattern extends valueof string = "^[a-zA-Z0-9-]{3,24}$"
+  NamePattern extends valueof string = "^[a-zA-Z0-9-]{3,24}$",
+  NameType extends string = string
```

Comment:
should it just be `Type`

Reply:
ARM mandates the name need to be string.

Reply:
I mean `NameType` param should be called `Type`?